templates/systems/controls_selected.html

```diff
@@ -20,9 +20,22 @@
 {% block body_content %}
   <div class="systems-top">
     <div id="tab-content" class="row rows-header">
-        <div id="" class="col-xs-2 col-sm-2 col-md-2 col-lg-2 col-xl-2">Selected controls</div>
-        <div id="" class="col-xs-9 col-sm-9 col-md-9 col-lg-9 col-xl-9">&nbsp;</div>
-        <div id="" class="col-xs-1 col-sm-1 col-md-1 col-lg-1 col-xl-1">
+        <div id="" class="col-xs-4 col-sm-4 col-md-4 col-lg-4 col-xl-4"><span class="systems-selected-components">Selected controls</span></div>
+        <div id="" class="col-xs-7 col-sm-7 col-md-7 col-lg-7 col-xl-7">
+          <form id="add-control" class="form-inline pull-right" method="post"
+                action="{%  url 'system_controls_add' system_id=system.id %}">
+            {% csrf_token %}
+            <div class="form-group">
+                <label for="control_id">Add control</label>&nbsp;&nbsp;
+                <!-- NOTE: Selection box converts to a Select2 selection box via script at bottom of page -->
+                <select class="producer_element_id" id="add-control_id-select" name="control_id" style="width:440px;">
+                    <option value="None" selected disabled hidden>Select a Control</option>
+                </select>
```

Comment:
```suggestion
                <select class="producer_element_id" id="add-control_id-select" name="control_id_catalogkey" onchange="$(form).submit();" style="width:440px;">
                    <option value="None" selected disabled hidden>Select a Control</option>
                                {% for control in controls %}
{#                                    {% if control.oscal_ctl_id not in controls %}#}
                                        <option value="{{ control.id }}-_-{{ control.oscal_catalog_key }}">{{ control.oscal_ctl_id }} {{ control.oscal_catalog_key }}</option>
{#                                    {% endif %}#}
                                {% endfor %}
                </select>
```
The controls selected dropdown is not fixed. it is not working because you are not listing controls to select... refer to element_detail_tabs.html. Here's a suggestion for this part. Something like this tied with the suggestion to views.py

Reply:
The controls select dropdown currently works differently from other drop downs by intentionally not displaying any values. The controls select drop down does not display any values (currently) because 3 currently have 3 control catalogs (800-171, 800-53 rev 4, 800-53 rev5) totaling over 2,000 controls -- too long for a drop down. So the controls select drop down is transformed into a select2 drop down by javascript and then does autocomplete from user typing to retrieve likely control matches from the server. The current performance of that retrieval is slow (about 2 seconds) making it appear like the drop down is broken. But the performance of the search in `/controls/api/controlsselect/` needs to be optimized (and made case insensitive).

The use of `-_-` string and passing a single field value on the form submissions is a nice improvement that needs to be implemented in the `/controls/api/controlsselect/` content returned on the Ajax autocomplete. Since both `controls_selected.html` and `element_detail_tabs.html` are using the same `/controls/api/controlsselect/`, I didn't want to make this change in this PR. I think it would be better to commit this PR and then optimize `/controls/api/controlsselect/` to better support both `controls_selected.html` and `element_detail_tabs.html`